.github/workflows/sphinx-python-dependency-build-checks.yml

```diff
@@ -0,0 +1,51 @@
+# The purpose of this workflow file is to confirm that the Sphinx
+# virtual environment can be built from source, consequently documenting
+# the packages required in the build environment to do that.
+#
+# This is needed because some projects embeds the documentation into built
+# artifacts which involves rendering the documentation on the target
+# architecture.
+#
+# Depending on the architecture, pip may or may not have already built wheels
+# available, and as such we need to make sure building wheels from source can
+# succeed.
+name: Sphinx python dependency build checks
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+  workflow_dispatch: # manual trigger
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Install dependencies
+      run: |
+        set -ex
+        sudo apt-get --fix-missing update
+        sudo apt -y install \
```

Comment:
Nitpick but you used `apt-get` the line before and it is preferred in scripts over `apt`.